sdks/java/extensions/sql/payloads/build.gradle

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins { id 'org.apache.beam.module' }
+
+applyJavaNature(
+        automaticModuleName: 'org.apache.beam.sdk.extensions.sql.meta.provider.payloads',
+        generatedClassPatterns: [
+                /^org\.apache\.beam\.sdk\.extensions\.sql\.meta\.provider\.kafka\.KafkaMessages/,
+                /^org\.apache\.beam\.sdk\.extensions\.sql\.meta\.provider\.kafka\.thrift.*/,
+        ],
+)
+
+applyGrpcNature()
+
+description = "Apache Beam :: SDKs :: Java :: Extensions :: SQL :: Payloads"
+ext.summary = "Byte array payload handling extensions for Beam SQL."
```

Comment:
Instead of creating a new module, could we move the PayloadSerializerProvider implementations to their respective modules?

IIUC, then they should get pulled in when a user adds those modules as dependencies

Reply:
Fixed